docs/root/configuration/http_filters/fault_filter.rst

```diff
@@ -36,6 +27,44 @@ Configuration
 * :ref:`v2 API reference <envoy_api_msg_config.filter.http.fault.v2.HTTPFault>`
 * This filter should be configured with the name *envoy.fault*.
 
+.. _config_http_filters_fault_injection_http_header:
+
+Controlling fault injection via HTTP headers
+--------------------------------------------
+
+The fault filter has the capability to allow fault configuration to be specified by the caller.
+This is useful in certain scenarios in which it is desired to allow the client to specify its own
+fault configuration. The currently supported header controls are:
+
+* Request delay configuration via the *x-envoy-fault-delay-request* header. The header value
+  should be an integer that specifies the number of milliseconds to throttle the latency for.
+* Response rate limit configuration via the *x-envoy-fault-throughput-response* header. The
+  header value should be an integer that specified the limit in KiB/s and must be > 0.
+
+.. attention::
+
+  Allowing header control is inherently dangerous if exposed to untrusted clients. In this case,
+  it is suggested to use the :ref:`max_active_faults
+  <envoy_api_field_config.filter.http.fault.v2.HTTPFault.max_active_faults>` setting to limit the
+  maximum concurrent faults that can be active at any given time.
+
+The following is an example configuration that enables header control for both of the above
+options:
+
+.. code-block:: yaml
+
+  name: envoy.fault
+  config:
+    max_active_faults: 100
+    delay:
+      header_delay: {}
+      percentage:
+        numerator: 100
+    response_rate_limit:
+      header_limit: {}
+      percentage:
+        numerator: 100
+
 .. _config_http_filters_fault_injection_runtime:
 
 Runtime
```

Comment:
Sorry, I may not have been clear on my last comment - I think the runtime overrides could use clarification in docs proper, not just in APIs.  I think we could make it more clear that overrides only work for features which are already configured.  i.e. you can't take a fault filter configured for aborts via filter config and runtime-inject delays.

It wasn't really necessary when the fault filter only did one thing, but now that it does a few, I could see someone getting confused